packages/core/src/observable/types.ts

```diff
@@ -0,0 +1,35 @@
+export type ObserverNext<T> = (value: T) => void
+export type ObserverError = (e: any) => void
+export type ObserverComplete = () => void
+
+export interface Observer<T> {
+  next: ObserverNext<T>
+  error: ObserverError
+  complete: ObserverComplete
+}
+
+/**
+ * An observable that aligns with the
+ * [TC39 observable proposal](https://github.com/tc39/proposal-observable) and
+ * can be consumed by other observable libraries like
+ * [rxjs](https://github.com/ReactiveX/rxjs) or
+ * [zen-observable](https://github.com/zenparsing/zen-observable).
+ */
+export interface Observable<T> {
+  subscribe(): Subscription
+  subscribe(observer: Partial<Observer<T>>): Subscription
+  subscribe(next: ObserverNext<T>): Subscription
+  subscribe(
+    next: ObserverNext<T>,
+    error: ObserverError,
```

Comment:
error is optional, so it should be  `error?: ObserverError` in order to align with tc39; I can also imagine that all params are optional, so that the interface is just  
```js
subscribe(observer: Partial<Observer<T>>): Subscription
subscribe(
    next?: ObserverNext<T>,
    error?: ObserverError,
    complete?: ObserverComplete
  ): Subscription
```

Reply:
if you dont pass error the previous signature matches:
`subscribe(next: ObserverNext<T>): Subscription`

there should be "no holes" (undefineds) in the arg list, which your proposed signature would allow:
```ts
subscribe(
    next?: ObserverNext<T>,
    error?: ObserverError,
    complete?: ObserverComplete
  ): Subscription

subscribe(undefined, undefined, () => console.log('yay')) // this is not required by the spec and deprecated in rxjs
```

Reply:
https://tc39.es/proposal-observable/#observable-prototype-subscribe

its actually a little bit confusing:
the described algorithmn allows everything in the first position:
- object -> observer, ignore rest of the args
- callable -> its the next function, check other "optional" args (so `subscribe(console.log, undefined, () => console.log('yay'))` is allowed)
- everything else -> empty observer (this includes `subscribe(undefined, undefined, () => console.log('yay'))`)

so insteadof:
```ts
export interface Observable<T> {
  subscribe(): Subscription
  subscribe(observer: Partial<Observer<T>>): Subscription
  subscribe(next: ObserverNext<T>): Subscription
  subscribe(
    next: ObserverNext<T>,
    error: ObserverError,
    complete?: ObserverComplete
  ): Subscription
}
```
its actually
```ts
export interface Observable<T> {
  subscribe(): Subscription
  subscribe(observer: Partial<Observer<T>>): Subscription
  subscribe(
    next: ObserverNext<T>,
    error?: ObserverError,
    complete?: ObserverComplete
  ): Subscription
}
```

Reply:
so it is then identical to what I suggested
```
subscribe(observer: Partial<Observer<T>>): Subscription
subscribe(
    next?: ObserverNext<T>,
    error?: ObserverError,
    complete?: ObserverComplete
  ): Subscription
```

Reply:
i did go with
```ts
  subscribe(observer: Partial<Observer<T>>): Subscription
  subscribe(
    next: ObserverNext<T>,
    error?: ObserverError,
    complete?: ObserverComplete
  ): Subscription
```

- the first parameter is required
- the first parameter can be anything, but in TS we can be more smart and only allow stuff that makes sense

```ts
subscribe(
    next?: ObserverNext<T>,
    error?: ObserverError,
    complete?: ObserverComplete
  ): Subscription
```
while its a valid signature according to the spec, it would produce an empty observer, when calling `subscribe(undefined, console.log)`, which is unexpected and therefore disallowed